### PR TITLE
feature: add logic to password reset

### PR DESCRIPTION
**Pull request by SergKukh**
What to check:
- create userApi
- add logic to PasswordResetRequest and PasswordReset pages
- add errors handling from backend

also fix import order and it does autofix in ProfileQuestions1

image

image

### CODE REVIEW

File: apps/freelance/src/pages/PasswordReset/index.tsx [hunk 1]
```diff
@@ -85,9 +83,25 @@ const PasswordReset = () => {
...
             {t('resetPassword.buttonText')}
           </Button>
+          {isError && (
+            <StyledError>
+              {t(
+                errorMessages.find(
+                  obj =>
+                    obj.error ===
+                    JSON.parse(JSON.stringify(error))?.data?.message,
+                )?.message || 'resetPassword.errors.unexpected',
```
Comment (lyudad) [inline apps/freelance/src/pages/PasswordReset/index.tsx]: its look weard, if all this transformation needed, create some function with name that will describe what you actually do there

File: apps/freelance/src/pages/PasswordResetRequest/constants.ts [hunk 2]
```diff
@@ -1,2 +1,7 @@
 export const emailName = 'email';
 export const infoFontSize = '1.1rem';
+export const errorFontSize = '0.9rem';
+export const errorColor = '#c93636';
```
Comment (lyudad) [inline apps/freelance/src/pages/PasswordResetRequest/constants.ts]: all colors should be in common file for full project, not for one component

File: apps/freelance/src/pages/PasswordResetRequest/index.tsx [hunk 3]
```diff
@@ -13,20 +14,18 @@ type Inputs = {
 const PasswordResetRequest = () => {
   const { t } = useTranslation();
   const { handleSubmit, control } = useForm<Inputs>();
-  const [loading, setLoading] = useState<boolean>(false);
-  const [email, setEmail] = useState<string>('');
+  const email = useRef<string>('');
```
Comment (lyudad) [inline apps/freelance/src/pages/PasswordResetRequest/index.tsx]: why refs needed, I dont like refs, if you use it - you should have good reason

File: apps/freelance/src/pages/PasswordReset/index.tsx (hunk 1 above)
Comment (SergKukh) [reply apps/freelance/src/pages/PasswordReset/index.tsx]: @lyudad fixed

File: apps/freelance/src/pages/PasswordResetRequest/constants.ts (hunk 2 above)
Comment (SergKukh) [reply apps/freelance/src/pages/PasswordResetRequest/constants.ts]: @lyudad fixed

File: apps/freelance/src/pages/PasswordResetRequest/index.tsx (hunk 3 above)
Comment (SergKukh) [reply apps/freelance/src/pages/PasswordResetRequest/index.tsx]: @lyudad change it to useState

**Review by lyudad — APPROVED** [review]

**Pull request merged**